From d9c4b22e74008cfdcb952b6fde0dde303b1259bb Mon Sep 17 00:00:00 2001
From: Thomas Tai <thomas.tai@windriver.com>
Date: Fri, 2 Oct 2009 15:17:28 -0400
Subject: [PATCH 23/23] ptrace: usermode-agent causing trap

When running usermode-agent -V, failed at PTRACE_SINGLESTEP.
Refer to wrlinux 2.0 hypervisor implementation and
found PTRACE_SINGLESTEP should be handled by default case
statement.

Signed-off-by: Thomas Tai <thomas.tai@windriver.com>
---
 kernel/ptrace.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/kernel/ptrace.c b/kernel/ptrace.c
index 85c1983..af2c195 100644
--- a/kernel/ptrace.c
+++ b/kernel/ptrace.c
@@ -445,9 +445,12 @@ int ptrace_request(struct task_struct *child, long request,
 		ret = ptrace_detach(child, data);
 		break;
 
+#ifndef CONFIG_PARAVIRT
 #ifdef PTRACE_SINGLESTEP
 	case PTRACE_SINGLESTEP:
 #endif
+#endif
+
 #ifdef PTRACE_SINGLEBLOCK
 	case PTRACE_SINGLEBLOCK:
 #endif
-- 
1.6.3.3

